File: 1/twosum.cpp
class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        unordered_map<int, int> myMap;

        for (int i = 0; i < nums.size(); i++)
        {
            if (myMap.find(target - nums[i]) != myMap.end())
            {
                return {i, myMap[target - nums[i]]};
            }
            else
            {
                myMap[nums[i]] = i;
            }
        }
        return {};
    };
};

File: Two sum/twoSum.js
function twoSum(nums, target) {
  const map = new Map();
  
  for (let i = 0; i < nums.length; i++) {
    const diff = target - nums[i];
    
    if (map.has(diff)) {
      return [i, map.get(diff)];
    }
    
    map.set(nums[i], i);
  }
}
